src/ga/ga_selectors.rs

```diff
@@ -315,15 +297,22 @@ impl<'a, T: GASolution> GASelector<'a, T> for GARouletteWheelSelector<'a, T>
         let mut upper = wheel_slots-1;
         let mut i;
 
-        while upper >= lower
+        while upper > lower
         {
             i = lower + (upper-lower)/2;
 
             assert!(i < wheel_slots);
 
             if self.wheel_proportions[i] > cutoff
             {
-                upper = i-1;
```

Comment:
This was causing intermittent panics btw, if i = 0, i-1 would underflow as i is unsigned.